src/main/scala/com/cognite/sdk/scala/v1/resources/files.scala

```diff
@@ -158,33 +148,22 @@ class Files[F[_]: Applicative](val requestSession: RequestSession[F])
         )
       )
 
-  def download(item: FileDownload, out: java.io.OutputStream): F[Unit] = {
-    val link: F[FileDownloadLink] = downloadLink(item)
-
-    requestSession.flatMap(
-      link,
-      (link: FileDownloadLink) => {
-        val response = requestSession.send { request =>
-          request
-            .get(
-              uri"${link.downloadUrl}"
-            )
-            .response(asByteArray)
-        }
-        requestSession.map(
-          response,
-          (res: Response[Either[String, Array[Byte]]]) =>
-            res.body match {
-              case Right(bytes) => out.write(bytes)
-              case Left(_) =>
-                throw SdkException(
-                  s"File download of file ${item.toString} failed with error code ${res.code.toString}"
-                )
-            }
-        )
+  def download(item: FileDownload, out: java.io.OutputStream)(implicit F: Sync[F]): F[Unit] =
+    for {
+      link <- downloadLink(item)
+      res <- requestSession.send { request =>
+        request
+          .get(uri"${link.downloadUrl}")
+          .response(asByteArray)
       }
-    )
-  }
+      bytes <- F.fromOption(
+        res.body.toOption,
+        SdkException(
+          s"File download of file ${item.toString} failed with error code ${res.code.toString}"
+        )
+      )
+      r <- F.blocking(out.write(bytes))
```

Comment:
actually looks like here we might as well just return F[ByteArray] since we read it all
but F.blocking works too
Or making it stream the body, but that all can be looked at separately as PR doesn't introduce new blocks :+1:

Reply:
good point! yep, I think ideally it should be streamed